src/utilities.ts

```diff
@@ -179,3 +184,25 @@ export function calculateProRatedUndernameCost(
       : fullCost;
   return proRatedCost;
 }
+
+export function calculateUndernamePermutations(domain: string): number {
+  const numberOfPossibleCharacters = 38; // 26 letters + 10 numbers + - (dash) + _ (underscore)
+  const numberOfAllowedStartingAndEndingCharacters = 36; // 26 letters + 10 numbers
+  const nameSpaceLength = NAMESPACE_LENGTH - domain.length; // should be between 11 and 61
+  let numberOfPossibleUndernames = 0;
+
+  for (
+    let undernameLength = 1;
+    undernameLength <= nameSpaceLength;
+    undernameLength++
+  ) {
+    if (undernameLength === 1 || undernameLength === nameSpaceLength) {
+      numberOfPossibleUndernames +=
+        numberOfAllowedStartingAndEndingCharacters ** undernameLength;
+    } else {
+      numberOfPossibleUndernames +=
+        numberOfPossibleCharacters ** undernameLength;
+    }
+  }
+  return numberOfPossibleUndernames;
+}
```

Comment:
this is technically nice - but as mentioned not necessary. we can use `MAX_ALLOWED_UNDERNAMES`

Reply:
I kept the function in there, also left it in the tests, but updated the check to use max_allowed_undernames